src/blockchain/transactions.rs

```diff
@@ -102,18 +136,20 @@ impl Transactions for BtcAnchoringService {
 
         // Check that input signature is correct.
         let redeem_script = actual_config.redeem_script();
-        let input_signer = InputSigner::new(redeem_script.clone());
+        let quorum = redeem_script.content().quorum;
+        let input_signer = InputSigner::new(redeem_script);
         arg.verify_signature(&input_signer, &public_key, &proposal, &expected_inputs)?;
 
         // All preconditions are correct and we can use this signature.
         let input_id = TxInputId::new(proposal.id(), arg.input);
-        let mut input_signatures = schema.input_signatures(&input_id, &redeem_script);
-        let quorum = redeem_script.content().quorum;
-        let mut input_signature_len = input_signatures.len();
+        let mut input_signatures = schema.input_signatures(&input_id);
+        let mut input_signature_len = input_signatures.0.len();
         // Check that we have not reached the quorum yet, otherwise we should not do anything.
         if input_signature_len < quorum {
             // Add signature to schema.
-            input_signatures.insert(anchoring_node_id, arg.input_signature.clone().into());
+            input_signatures
+                .0
```

Comment:
I think it will be really nice to add several methods to `impl BinaryMap` to hide this `.0` field usage, since it's very counter-intuitive.